xls/dslx/lsp/language_server_adapter.cc

```diff
@@ -421,4 +421,39 @@ LanguageServerAdapter::Rename(std::string_view uri,
   return std::nullopt;
 }
 
+absl::StatusOr<std::vector<verible::lsp::DocumentHighlight>>
+LanguageServerAdapter::DocumentHighlight(
+    std::string_view uri, const verible::lsp::Position& position) const {
+  if (ParseData* parsed = FindParsedForUri(uri); parsed && parsed->ok()) {
+    FileTable& file_table = parsed->file_table();
+    const Pos pos = ConvertLspPositionToPos(uri, position, file_table);
+    VLOG(1) << "FindDefinition; uri: " << uri << " pos: " << pos;
+    const Module& module = parsed->module();
+    std::optional<const NameDef*> maybe_definition = xls::dslx::FindDefinition(
+        module, pos, parsed->type_info(), parsed->import_data());
+    if (maybe_definition.has_value()) {
+      const NameDef* name_def = maybe_definition.value();
+      const Span& definition_span = name_def->span();
+      std::vector<verible::lsp::DocumentHighlight> highlights = {
+          verible::lsp::DocumentHighlight{
+              .range = ConvertSpanToLspRange(definition_span),
+              // TODO(cdleary): 2024-10-13 How do we indicate the kind?
```

Comment:
Currently there is none defined in the verible lsp stuff, so we just default to 'text'.
If we have meaningful read/write annotations we can make (that hopefully will show up in a useful way in editors), we can add it to the protocol schema where this TODO is currently:

 https://github.com/chipsalliance/verible/blob/master/common/lsp/lsp-protocol.yaml#L152

... and then adding the enum here
https://github.com/chipsalliance/verible/blob/master/common/lsp/lsp-protocol-enums.h

Reply:
Understood -- not sure how the editor does/would show them differently so just removed them for now.